include/dds/core/xtypes/detail/MemberType.hpp

```diff
@@ -18,12 +18,195 @@
 #ifndef EPROSIMA_DDS_CORE_XTYPES_DETAIL_MEMBER_TYPE_HPP_
 #define EPROSIMA_DDS_CORE_XTYPES_DETAIL_MEMBER_TYPE_HPP_
 
+#include <dds/core/xtypes/Annotation.hpp>
+#include <dds/core/xtypes/DynamicType.hpp>
+
+#include <vector>
+#include <functional>
+#include <algorithm>
+
 namespace dds {
 namespace core {
 namespace xtypes {
 namespace detail {
 
-class MemberType { };
+class MemberType
+{
+public:
+    MemberType(
+            const std::string &name,
+            xtypes::DynamicType &dt):
+        name_(name),
+        dt_(dt.name(),dt.kind(), dt.annotations()),
+        ann_()
+    {
+    }
+
+    MemberType(
+            const std::string &name,
+            xtypes::DynamicType &dt,
+            xtypes::Annotation &a):
+        name_(name),
+        dt_(dt.name(),dt.kind(), dt.annotations()),
+        ann_()
+    {
+        ann_.push_back(a) ;
+    }
+
+    void name(
+            const std::string &name)
+    {
+        name_ = name ;
+    }
+
+    void dt(
+            const xtypes::DynamicType &dt)
+    {
+        dt_ = dt ;
+    }
+
+    void annotation(
+            std::vector<xtypes::Annotation> &ann)
+    {
+        ann_.reserve(ann.size() + ann_.size()) ;
+        for (auto it = ann.begin() ; it != ann.end() ; ++it)
+        {
+            ann_.emplace_back(*it) ;
+        }
+    }
+
+    template<typename AnnoIter>
+    void annotation(
+            AnnoIter begin,
+            AnnoIter end)
+    {
+        ann_.reserve(ann_.size() + ( end - begin) ) ;
+        for (auto it = begin ; it != end ; ++it)
+        {
+            ann_.emplace_back(*it) ;
+        }
+    }
+
+    void annotation(
+            xtypes::Annotation &ann)
+    {
+        ann_.push_back(ann)  ;
+    }
+
+    const std::string &name()const noexcept
+    {
+        return name_ ;
+    }
+
+    const xtypes::DynamicType &dt() const noexcept
```

Comment:
Personally, the abbreviations are ok for local variables inside of functions, but for public functions I would write the entire name in order to be more verbose to the user.

Reply:
Fine, I'll put a more verbose interface